src/workerd/api/pyodide/setup-emscripten.c++

```diff
@@ -0,0 +1,80 @@
+#include "setup-emscripten.h"
+
+#include <pyodide/generated/emscripten_setup.capnp.h>
+
+namespace workerd::api::pyodide {
+
+jsg::JsValue SetupEmscripten::getModule(jsg::Lock& js) {
+  auto jsval = pyodide::initializeEmscriptenRuntime(js);
+  module_ = jsval.addRef(js);
+  return KJ_ASSERT_NONNULL(module_).getHandle(js);
+}
+
+v8::Local<v8::Module> loadEmscriptenSetupModule(jsg::Lock& js) {
+  v8::Local<v8::String> contentStr =
+      jsg::v8Str(js.v8Isolate, EMSCRIPTEN_SETUP->getCode().asArray());
+  v8::ScriptOrigin origin(
+      jsg::v8StrIntern(js.v8Isolate, "pyodide-internal:generated/emscriptenSetup"), 0, 0, false, -1,
+      {}, false, false, true);
+  v8::ScriptCompiler::Source source(contentStr, origin);
+  return jsg::check(v8::ScriptCompiler::CompileModule(js.v8Isolate, &source));
+}
+
+jsg::JsValue resolvePromise(jsg::Lock& js, jsg::JsValue prom) {
+  auto promise = KJ_ASSERT_NONNULL(prom.tryCast<jsg::JsPromise>());
+  if (promise.state() == jsg::PromiseState::PENDING) {
+    js.runMicrotasks();
+  }
+  KJ_ASSERT(promise.state() == jsg::PromiseState::FULFILLED);
+  return promise.result();
+}
```

Comment:
Interesting. So this is asserting that each promise will be resolved in one trip around the event loop or... something?

Reply:
```c++
  /**
   * Runs the default MicrotaskQueue until it gets empty and perform other
   * microtask checkpoint steps, such as calling ClearKeptObjects. Asserts that
   * the MicrotasksPolicy is not kScoped. Any exceptions thrown by microtask
   * callbacks are swallowed.
   */
```
So it's not one trip around the event loop but rather, until we're completely done processing any events.

Reply:
But only events that are microtasks not the main event loop?

Reply:
What do you believe we are running here that does not count as a microtask?

Reply:
I don't expect network IO to work if that's the question

Reply:
What about instantiating webassembly?

Reply:
Well it works so I guess it is 😁

Reply:
Good enough for me =)